src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -474,23 +475,35 @@ private async Task SyncTriggers(Site functionApp)
             }, retryCount: 5);
         }
 
-        private async Task<bool> HandleElasticPremiumLinuxPublish(Site functionApp, Func<Task<Stream>> zipStreamFactory)
+        private async Task HandleElasticPremiumLinuxPublish(Site functionApp, Func<Task<Stream>> zipStreamFactory)
         {
+            // If Azure Files settings are present, we deploy similar to Linux App Service
+            if (functionApp.AzureAppSettings.ContainsKey(Constants.WebsiteContentAzureFileConnectionString)
+                && functionApp.AzureAppSettings.ContainsKey(Constants.WebsiteContentShared))
+            {
+                await HandleLinuxAppServicePublish(functionApp, zipStreamFactory);
+                return;
+            }
+
+            ColoredConsole.WriteLine(WarningColor($"Your Linux Premium function app does not have Azure Files Application Settings" +
+                $" ('{Constants.WebsiteContentAzureFileConnectionString}', '{Constants.WebsiteContentShared}') configured. Azure Files setup is required to enable" +
+                $"persistent storage in Linux Premium function apps." +
+                $" For more info, see https://aka.ms/linux-ep-storage"));
+
+            // If Azure Files settings are not present, we need to do Run From Package = URL deployment
```

Comment:
I thought local zip can work with AppService storage as well?

Reply:
@balag0, in Premium scenarios with cross-stamp scaling in play, the storage in one instance may not be same as the storage in another instance of the same app (If not using Azure files, and relying on the stamp based storage). So, an instance in a different stamp may not be able to access the app content.
Using Run From Package = URL at least ensures that the apps continue to work in those case.

Reply:
Yes, But if the app is not configured with AzureFiles it wont be eligible for cross stamp scale in the first place?